include/bluetooth/mesh/main.h

```diff
@@ -649,6 +649,13 @@ struct bt_mesh_friend_cb {
  */
 int bt_mesh_friend_terminate(uint16_t lpn_addr);
 
+/** @brief Store pending RPL entries in the persistent storage.
+ *
+ * This API allows the user to store pending RPL entries in the persistent
+ * storage without waiting for the store timeout.
+ */
+void bt_mesh_rpl_pending_store(void);
```

Comment:
Could this take an address as a parameter perhaps? This way, we'd have control over which entries are stored, and could selectively store the RPL entry for critical messages right away, in case the timeout is long. Passing `BT_MESH_ADDR_ALL_NODES` could keep the current behavior.

This would change the mechanism in rpl.c a bit, perhaps it should just bypass the timer mechanism in this case?

If we end up changing the storage scheme in the future to store all addresses together, we might not be able to provide this granularity of control, but that'd be acceptable IMO.

Reply:
IMO this is not a good idea. This seems like an optimization however end application should not predict what is important and what is not at the level of source address in RPL as that gives rise to vulnerability. Also, there is no interoperable way to specify such an address in a mesh.

Reply:
> end application should not predict what is important and what is not at the level of source address in RPL as that gives rise to vulnerability.

I believe this would have the exact opposite effect. We're already running RPL on a timer, so there's already a potential for failing to store RPL before power down. Before this PR, we had no way to even prevent it, apart from setting the timer to 0, which isn't a realistic option for most storage configurations. This API solves this problem, and I'm arguing that the API should be as good as possible.

For instance, if my device supports listening to periodic sensor input from 50 sensors, as well as locking/unlocking a door, I can set the RPL storage timer to e.g. 120 seconds, so I don't wear out my flash to guarantee protection for the sensor messages. If I receive a message that unlocks the door, I would like to store this right away, so a sudden power cut wouldn't expose this functionality for replay attacks. I have already decided that storing the RPL entries from the sensors is low priority for me, so I only really care about putting the door unlock message into persistent storage. If this sort of mechanism shouldn't be allowed, this entire PR is invalid IMO, and the existing RPL storage timer should be removed.

Being able to store just the critical entry would not change the security aspect of this API, but it would make this a better API.

I don't think we should be nerfing the API to protect uninformed users, when there is a real use case for this API for users that know what they're doing and the API doesn't introduce complexity for other use cases. If users want to shoot themself in the foot, there are plenty of opportunities for this already (e.g. the default RPL timer value), and we can't really protect them. The API documentation should inform users of the risk and responsibility their application takes on if they decide to forego the timer, but if we want to have this API, it might as well be as good as possible.

> Also, there is no interoperable way to specify such an address in a mesh.

I don't understand why this needs to be interoperable, could you please elaborate? This API is not meant to be exposed to other devices.